### PR TITLE
Use -Wall, -Wextra, remove warnings, deprecate unused executables including alchemy and ccstoh5

### PR DESCRIPTION
**Pull request by ylipacbio**
Use -Wall -Wextra  in makefile
Remove -fpermissive and -Wuninitialized, and -Wno-div-by-zero
Remove all warnings
Deprecate unsupported executables: ccstoh5 (replaced by bax2bam) and alchemy (replaced by pbalchemy)

### CODE REVIEW

File: extrautils/BasH5Simulator.cpp [hunk 1]
```diff
@@ -1,524 +0,0 @@
-#include <string>
```
Comment (pb-cdunn) [inline extrautils/BasH5Simulator.cpp]: I like deleted code! :haircut:

File: extrautils/makefile [hunk 2]
```diff
@@ -5,12 +5,12 @@ SRCDIR:=$(dir $(realpath $(lastword $(MAKEFILE_LIST))))
 include ${SRCDIR}/../rules.mk
 
 CXXOPTS := -std=c++0x -pedantic \
-           -Wall -Wuninitialized -Wno-div-by-zero \
```
Comment (pb-cdunn) [inline extrautils/makefile]: Hmm. I don't think `-Wall -Wextra` includes `-Wuninitialized`.
* https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#Warning-Options

`-Wno-div-by-zero` is typically used to allow a cheat for Infinity. Is that done somewhere now?

File: extrautils/makefile [hunk 3]
```diff
@@ -5,12 +5,12 @@ SRCDIR:=$(dir $(realpath $(lastword $(MAKEFILE_LIST))))
...
-           -MMD -MP -w -fpermissive
+           -Wall -Wextra \
+           -MMD -MP -w
```
Comment (pb-cdunn) [inline extrautils/makefile]: `-w` (lower-case) inhibits all warnings. I don't know what happens when it follow `-Wall`, but is it really needed here?

Comment (ylipacbio) [inline extrautils/makefile]: Let me remove -w. This may have prevented all warnings from been displayed!

Comment (ylipacbio) [inline extrautils/makefile]: Haha, I saw lots of warnings since -w was removed. Fixing them now.

blasr_libcpp pull request 110 also need to be updated.

**Pull request merged**